src/entity/account/user.php

```diff
@@ -228,7 +228,10 @@ public function get_recover()
             return null;
         }
 
-        return substr(hash_hmac('crc32b', $this->recover_time, $this->id), 0, 8);
+        $token_hash = config('engine.entity.account.tokenhash', 'crc32b');
+        $token_length = config('engine.entity.account.tokenlen', 8);
+
+        return substr(hash_hmac($token_hash, $this->recover_time, $this->id), 0, $token_length);
```

Comment:
Seems the `hash` function still supports crc32b mode, why not using it to avoid making this code more complex that required? http://php.net/manual/fr/function.hash.php